File: demo-application/src/main/java/de/jds/DemoPersistenceCallback.java
package de.jds;

import org.springframework.sync.diffsync.PersistenceCallback;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class DemoPersistenceCallback implements PersistenceCallback<String> {

  final Map<String, String> store = new HashMap<>();

  @Override
  public List<String> findAll() {
    return new ArrayList<>(store.values());
  }

  @Override
  public String findOne(final String s) {
    return store.get(s);
  }

  @Override
  public void persistChange(final String s) {
  }

  @Override
  public void persistChanges(final List<String> list, final List<String> list1) {

  }

  @Override
  public Class<String> getEntityType() {
    return String.class;
  }
}

File: demo-application/src/main/kotlin/de/jds/example/DemoPersistenceCallback.kt
package de.jds.example

import de.sync.core.persistence.PersistenceCallback

class DemoPersistenceCallback : PersistenceCallback<Todo> {
    private val store: MutableMap<String, Todo> = mutableMapOf()
    override fun findAll(): List<Todo> {
        return ArrayList(store.values)
    }

    override fun findOne(id: String?): Todo? {
        return store[id]
    }

    override fun persistChange(s: Todo) {
        store[s.id] = s
    }

    override fun persistChanges(
        itemsToSave: List<Todo>?,
        itemsToDelete: List<Todo>?
    ) {
        itemsToSave?.forEach {
            store[it.id] = it
        }

        itemsToDelete?.forEach {
            store.remove(it.id)
        }
    }

    override val entityType: KClass<Todo>
        get() = Todo::class
}